modules/core/src/main/java/org/apache/ignite/internal/processors/cache/version/GridCacheVersionEx.java

```diff
@@ -105,7 +107,7 @@ public GridCacheVersionEx(int topVer, int nodeOrderDrId, long order, GridCacheVe
         }
 
         switch (writer.state()) {
-            case 3:
+            case 4:
```

Comment:
Hate to say this but I believe that new version of object won't be serialized properly on old nodes when RU enabled.
It'll try to read field "drVer" but "updateCounter" is found. "u" goes after "d" in alphabet, so reader will assume that "drVer" field is removed and will nullify it. Have you tested it? Am I wrong?
We need to make sure that this code is 100% safe.

Reply:
Sounds sad. I believe it is safe to add new fields.
I'll check this.

Reply:
Cool, now it works for sure!